components/gfx/font_context.rs

```diff
@@ -206,16 +211,18 @@ impl<S: FontSource> FontContext<S> {
         font_descriptor: FontDescriptor,
         synthesized_small_caps: Option<FontRef>,
     ) -> Result<FontRef, &'static str> {
-        let font_instance_key = self
-            .font_source
-            .get_font_instance(font_template.identifier(), font_descriptor.pt_size);
-
-        Ok(Rc::new(RefCell::new(Font::new(
-            font_template,
-            font_descriptor,
-            font_instance_key,
+        let mut font = Font::new(
+            font_template.clone(),
+            font_descriptor.clone(),
             synthesized_small_caps,
-        )?)))
+        )?;
+        font.font_key = self.font_source.get_font_instance(
```

Comment:
We are first instantiating the `Font` with the default `FontInstanceKey` and then overriding it again with a valid one - is this because the a `Font` instance is needed to first get the flags to be passed to WebRender in order to get the key from it?

In the future and not necessarily in this same PR, would it be possible to get the flags from the template instead of the `Font` instance?

Reply:
I think it would be hard to get this from the `FontTemplate`, but there's a possibility that we could create the `PlatformFont` first and get it from there. The other option is we could move the `FontInstanceKey` out of the font or make it an `Option`. There could conceivably be some cases where we want to keep the font cached, but also not have it in WebRender.